.paw/work/agent-installation-management/Spec.md

```diff
@@ -0,0 +1,241 @@
+2. Given all agent files are present and version markers match, When extension activates, Then no file writes occur (idempotent behavior).
+3. Given agent file content was manually corrupted, When extension activates with version marker intact, Then the extension does not rewrite files (preserves user modifications until explicit version update).
+
+### User Story P5 – Agent Migration and Cleanup
+
+Narrative: PAW 0.4.0 renamed agent files from `.chatmode.md` to `.agent.md` extension and renamed "Planner" to "Planning". A developer upgrades from 0.3.0 to 0.4.0. During activation, the extension consults an internal migration manifest mapping old filenames to new ones. It installs the new `.agent.md` files, then removes obsolete `.chatmode.md` files based on known PAW naming patterns. The developer's prompts directory contains only current agent files without legacy clutter.
+
+Independent Test: Install PAW 0.3.0 agents, upgrade to 0.4.0, verify old `.chatmode.md` files removed and new `.agent.md` files installed.
+
+Acceptance Scenarios:
+1. Given PAW 0.3.0 installed `paw-planner.chatmode.md`, When upgrading to 0.4.0 with new `paw-planning.agent.md` filename, Then the extension installs new file and removes old file.
+2. Given multiple old agent files exist from previous versions, When extension activates with current version, Then all obsolete PAW-managed files are removed.
+
+### Edge Cases
+
+- **Multiple VS Code instances**: If multiple VS Code windows activate simultaneously and attempt to write agents concurrently, file system race conditions may occur. Expected behavior: Last writer wins; idempotent installation ensures eventual consistency across all instances.
+- **User manual modification of agent files**: If a user manually edits an installed agent file, the next version upgrade will overwrite their changes (because agents are treated as managed assets). Expected behavior: Changes lost on upgrade.
+- **Non-standard VS Code distributions**: VSCodium and Code-OSS use different configuration paths than standard VS Code. Expected behavior: Extension detects distribution-specific paths and installs to appropriate location for each variant.
+- **Disk full during installation**: If disk space is exhausted mid-write, partial files may be written. Expected behavior: File system error caught, error notification displayed, idempotent installation completes missing files on next activation.
+- **Copilot extension not installed**: If GitHub Copilot extension is not installed, agent files are written but unusable. Expected behavior: Agents installed successfully; no error (agents become available when user later installs Copilot).
+- **Profile switching**: VS Code supports multiple profiles, but user prompts directory is global (not per-profile). Expected behavior: Agents available across all profiles; no per-profile installation needed.
+
+## Requirements
```

Comment:
Uninstall is missing - if PAW gets uninstalled, it should remove the agents from the user directory so that they no longer appear in VS Code.